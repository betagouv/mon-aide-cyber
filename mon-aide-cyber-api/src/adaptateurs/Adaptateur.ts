export interface Adaptateur<T> {
  lis(): Promise<T>;
}
