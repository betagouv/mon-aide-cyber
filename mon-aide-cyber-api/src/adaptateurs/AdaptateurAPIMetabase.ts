import { AdaptateurMetabase, ReponseMetabase } from './AdaptateurMetabase';
import jwt from 'jsonwebtoken';

export class AdaptateurAPIMetabase implements AdaptateurMetabase {
  constructor(private readonly clefSecrete: string) {}

  appelle(): Promise<ReponseMetabase> {
    const token = jwt.sign(
      {
        resource: { dashboard: 5 },
        params: {},
        exp: Math.round(Date.now() / 1000) + 10 * 60,
      },
      this.clefSecrete
    );
    return Promise.resolve({
      corps: `${process.env.METABASE_URL}/embed/dashboard/${token}#bordered=true&titled=true`,
    });
  }
}
