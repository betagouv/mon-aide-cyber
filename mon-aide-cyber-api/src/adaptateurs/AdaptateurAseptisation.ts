import { RequestHandler } from 'express';

export interface AdaptateurAseptisation {
  aseptise(...champsAAseptiser: string[]): RequestHandler;
}
