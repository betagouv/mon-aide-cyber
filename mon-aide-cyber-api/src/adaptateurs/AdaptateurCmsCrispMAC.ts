import { CmsCrisp, PageHtmlCrisp } from '@lab-anssi/lib';
import { adaptateurEnvironnement } from './adaptateurEnvironnement';

export type Article = PageHtmlCrisp;

export interface AdaptateurCmsCrispMAC {
  recupereGuideAidantCyber(): Promise<Article>;

  promouvoirDiagnosticCyber(): Promise<Article>;

  promouvoirCommunauteAidantsCyber(): Promise<Article>;

  relaisAssociatifs(): Promise<Article>;
}

class AdaptateurCmsCrispMACConcret implements AdaptateurCmsCrispMAC {
  private cmsCrisp: CmsCrisp;

  constructor(idSite: string, clefAPI: string) {
    this.cmsCrisp = new CmsCrisp(idSite, clefAPI);
  }

  relaisAssociatifs(): Promise<PageHtmlCrisp> {
    throw new Error('Method not implemented.');
  }

  async promouvoirCommunauteAidantsCyber(): Promise<PageHtmlCrisp> {
    return await this.cmsCrisp.recupereArticle(
      adaptateurEnvironnement.crisp()?.promouvoirCommunauteAidantsCyber ?? ''
    );
  }

  async promouvoirDiagnosticCyber(): Promise<PageHtmlCrisp> {
    return await this.cmsCrisp.recupereArticle(
      adaptateurEnvironnement.crisp()?.promouvoirDiagnosticCyber ?? ''
    );
  }

  async recupereGuideAidantCyber(): Promise<Article> {
    return await this.cmsCrisp.recupereArticle(
      adaptateurEnvironnement.crisp()?.guideAidantCyber ?? ''
    );
  }
}

class AdaptateurCmsCrispMACMemoire implements AdaptateurCmsCrispMAC {
  private articeBouchone = {
    titre: 'Un article',
    contenu: 'Un contenu',
    description: 'Une description',
    tableDesMatieres: [],
  };

  async relaisAssociatifs(): Promise<PageHtmlCrisp> {
    return this.articeBouchone;
  }

  async promouvoirCommunauteAidantsCyber(): Promise<PageHtmlCrisp> {
    return this.articeBouchone;
  }

  async promouvoirDiagnosticCyber(): Promise<PageHtmlCrisp> {
    return this.articeBouchone;
  }
  async recupereGuideAidantCyber(): Promise<Article> {
    return this.articeBouchone;
  }
}

export const unAdaptateurCmsCrisp = (): AdaptateurCmsCrispMAC => {
  const configurationCrisp = adaptateurEnvironnement.crisp();
  if (!configurationCrisp) {
    return new AdaptateurCmsCrispMACMemoire();
  }
  return new AdaptateurCmsCrispMACConcret(
    configurationCrisp.idSite,
    configurationCrisp.clefAPI
  );
};
