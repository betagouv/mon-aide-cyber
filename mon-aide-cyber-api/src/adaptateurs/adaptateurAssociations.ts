import fs from 'fs';
import * as YAML from 'yaml';
import { Association } from '../api/associations/routesAssociations';

export const adaptateurAssociations = {
  referentiel: (cheminFichier: string): Association[] => {
    const contenuFichier = fs.readFileSync(cheminFichier, 'utf8');
    return YAML.parse(contenuFichier) as Association[];
  },
};
