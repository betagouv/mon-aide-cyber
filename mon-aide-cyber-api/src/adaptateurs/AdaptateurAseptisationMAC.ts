import { RequestHandler } from 'express';
import { AdaptateurAseptisation } from './AdaptateurAseptisation';
import { check } from 'express-validator';

export class AdaptateurAseptisationMAC implements AdaptateurAseptisation {
  aseptise(...champsAAseptiser: string[]): RequestHandler {
    return async (requete: any, _reponse: any, suite: any) => {
      const aseptisations = champsAAseptiser.map((p) =>
        check(p).trim().escape().run(requete)
      );
      await Promise.all(aseptisations);
      suite();
    };
  }
}
