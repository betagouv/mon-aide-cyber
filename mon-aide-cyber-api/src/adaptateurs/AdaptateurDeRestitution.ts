import {
  Indicateurs,
  ORDRE_THEMATIQUES,
  RecommandationPriorisee,
} from '../diagnostic/Diagnostic';
import { ContenuHtml } from '../infrastructure/adaptateurs/AdaptateurDeRestitutionPDF';
import { Restitution } from '../restitution/Restitution';

const estRecommandationPriorisee = (
  recommandationPriorisees: RecommandationPriorisee[] | undefined,
): recommandationPriorisees is RecommandationPriorisee[] => {
  return (
    recommandationPriorisees !== undefined &&
    recommandationPriorisees.length > 0
  );
};

export abstract class AdaptateurDeRestitution<T> {
  genereRestitution(restitution: Restitution): Promise<T> {
    const informations = this.genereInformations(restitution);
    const indicateursRestitution: Indicateurs = Object.entries(
      restitution.indicateurs,
    )
      .sort(([thematiqueA], [thematiqueB]) =>
        ORDRE_THEMATIQUES.indexOf(thematiqueA) >
        ORDRE_THEMATIQUES.indexOf(thematiqueB)
          ? 1
          : -1,
      )
      .reduce(
        (accumulateur, [thematique, indicateur]) => ({
          ...accumulateur,
          [thematique]: indicateur,
        }),
        {},
      );
    const indicateurs = this.genereIndicateurs(indicateursRestitution);
    const recommandations = this.genereMesuresPrioritaires(
      restitution.recommandations.recommandationsPrioritaires,
    );
    const autresRecommandations =
      restitution.recommandations.autresRecommandations;

    if (estRecommandationPriorisee(autresRecommandations)) {
      return this.genere([
        informations,
        indicateurs,
        recommandations,
        this.genereAutresMesures(autresRecommandations),
      ]);
    }
    return this.genere([informations, indicateurs, recommandations]);
  }

  protected abstract genereAutresMesures(
    autresRecommandations: RecommandationPriorisee[],
  ): Promise<ContenuHtml>;

  protected abstract genereMesuresPrioritaires(
    recommandationsPrioritaires: RecommandationPriorisee[] | undefined,
  ): Promise<ContenuHtml>;

  protected abstract genere(
    htmlRecommandations: Promise<ContenuHtml>[],
  ): Promise<T>;

  protected abstract genereIndicateurs(
    indicateurs: Indicateurs | undefined,
  ): Promise<ContenuHtml>;

  protected abstract genereInformations(
    restitution: Restitution,
  ): Promise<ContenuHtml>;
}
