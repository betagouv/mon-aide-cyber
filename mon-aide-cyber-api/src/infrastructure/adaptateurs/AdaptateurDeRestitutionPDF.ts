import { AdaptateurDeRestitution } from '../../adaptateurs/AdaptateurDeRestitution';
import * as pug from 'pug';
import puppeteer, { Browser, PDFOptions } from 'puppeteer';
import { PDFDocument } from 'pdf-lib';

export class AdaptateurDeRestitutionPDF extends AdaptateurDeRestitution<Buffer> {
  constructor(traductionThematiques: Map<string, string>) {
    super(traductionThematiques);
  }

  protected genere(mesures: Promise<ContenuHtml>[]) {
    return Promise.all(mesures)
      .then((htmls) => generePdfs(htmls))
      .then((pdfs) => fusionnePdfs(pdfs))
      .catch((erreur) => {
        console.log('Erreur génération recos', erreur);
        throw new Error(erreur);
      });
  }

  async genereHtml(pugCorps: string, paramsCorps: any): Promise<ContenuHtml> {
    const fonctionInclusionDynamique = (
      cheminTemplatePug: string,
      options = {}
    ) => {
      return pug.renderFile(
        `src/infrastructure/restitution/pdf/modeles/${cheminTemplatePug}`,
        options
      );
    };
    return Promise.all([
      pug.compileFile(
        `src/infrastructure/restitution/pdf/modeles/${pugCorps}.pug`
      )({
        ...paramsCorps,
        include: fonctionInclusionDynamique,
      }),
    ])
      .then(([corps]) => ({ corps, entete: '', piedPage: '' }))
      .catch((erreur) => {
        console.log('Erreur génération HTML', erreur);
        throw new Error(erreur);
      });
  }
}

const fusionnePdfs = (pdfs: Buffer[]): Promise<Buffer> => {
  const fusionne = (fusion: PDFDocument): Promise<PDFDocument> => {
    return pdfs.reduce(
      (pdfDocument, pdf) =>
        pdfDocument.then((f) =>
          PDFDocument.load(pdf)
            .then((document) =>
              f.copyPages(document, document.getPageIndices())
            )
            .then((copie) => copie.forEach((page) => f.addPage(page)))
            .then(() => pdfDocument)
        ),
      Promise.resolve(fusion)
    );
  };

  return PDFDocument.create()
    .then((fusion) => {
      return fusionne(fusion);
    })
    .then((fusion) => fusion.save())
    .then((pdf) => Buffer.from(pdf));
};

export type ContenuHtml = { corps: string; entete: string; piedPage: string };

const generePdfs = async (pagesHtml: ContenuHtml[]): Promise<Buffer[]> => {
  const pagesHtmlRemplies = pagesHtml.filter(
    (pageHtml) => pageHtml.corps !== ''
  );

  const navigateur = await lanceNavigateur();
  try {
    const contenu = pagesHtmlRemplies.reduce(
      (contenuPrecedent, contenuCourant) => {
        contenuPrecedent.corps += contenuCourant.corps;

        return contenuPrecedent;
      },
      {
        corps: pug.compileFile(
          `src/infrastructure/restitution/pdf/modeles/restitution.pug`
        )(),
      } as ContenuHtml
    );
    const contenuFinal: ContenuHtml = {
      corps: contenu.corps,
      entete: pug.compileFile(
        `src/infrastructure/restitution/pdf/modeles/restitution.entete.pug`
      )(),
      piedPage: pug.compileFile(
        `src/infrastructure/restitution/pdf/modeles/restitution.piedpage.pug`
      )(),
    };
    const resultat = navigateur.newPage().then((page) => {
      return page
        .setContent(contenuFinal.corps)
        .then(() =>
          page.pdf(formatPdfA4(contenuFinal.entete, contenuFinal.piedPage))
        )
        .catch((erreur) => {
          console.log(erreur);
          throw erreur;
        });
    });
    return await Promise.all([resultat]);
  } catch (erreur) {
    console.log(erreur);
    throw erreur;
  } finally {
    navigateur.close();
  }
};

const formatPdfA4 = (enteteHtml: string, piedPageHtml: string): PDFOptions => ({
  format: 'A4',
  printBackground: true,
  displayHeaderFooter: true,
  headerTemplate: enteteHtml,
  footerTemplate: piedPageHtml,
  margin: { bottom: '2cm', left: '1cm', right: '1cm', top: '23mm' },
});

const lanceNavigateur = (): Promise<Browser> =>
  puppeteer.launch({
    // Documentation des arguments : https://peter.sh/experiments/chromium-command-line-switches/
    // Inspiration pour la liste utilisée ici : https://www.bannerbear.com/blog/ways-to-speed-up-puppeteer-screenshots/
    // Le but est d'avoir un Puppeteer le plus léger et rapide à lancer possible.
    args: [
      '--no-sandbox',
      '--font-render-hinting=none',
      '--disable-gpu',
      '--autoplay-policy=user-gesture-required',
      '--disable-accelerated-2d-canvas',
      '--disable-background-networking',
      '--disable-background-timer-throttling',
      '--disable-backgrounding-occluded-windows',
      '--disable-breakpad',
      '--disable-client-side-phishing-detection',
      '--disable-component-update',
      '--disable-default-apps',
      '--disable-dev-shm-usage',
      '--disable-domain-reliability',
      '--disable-extensions',
      '--disable-features=AudioServiceOutOfProcess',
      '--disable-hang-monitor',
      '--disable-ipc-flooding-protection',
      '--disable-notifications',
      '--disable-offer-store-unmasked-wallet-cards',
      '--disable-popup-blocking',
      '--disable-print-preview',
      '--disable-prompt-on-repost',
      '--disable-renderer-backgrounding',
      '--disable-setuid-sandbox',
      '--disable-speech-api',
      '--disable-sync',
      '--hide-scrollbars',
      '--ignore-gpu-blacklist',
      '--metrics-recording-only',
      '--mute-audio',
      '--no-default-browser-check',
      '--no-first-run',
      '--no-pings',
      '--no-zygote',
      '--password-store=basic',
      '--single-process',
      '--use-gl=swiftshader',
      '--use-mock-keychain',
    ],
  });
