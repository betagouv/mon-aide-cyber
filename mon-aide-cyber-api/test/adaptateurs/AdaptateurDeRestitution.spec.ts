import { describe, it } from 'vitest';
import { AdaptateurDeRestitution } from '../../src/adaptateurs/AdaptateurDeRestitution';
import {
  genereLaRestitution,
  Indicateurs,
  RecommandationPriorisee,
} from '../../src/diagnostic/Diagnostic';
import { unDiagnostic } from '../constructeurs/constructeurDiagnostic';
import {
  uneListeDeQuestions,
  unReferentiel,
} from '../constructeurs/constructeurReferentiel';
import { unTableauDeRecommandations } from '../constructeurs/constructeurTableauDeRecommandations';
import { ContenuHtml } from '../../src/infrastructure/adaptateurs/AdaptateurDeRestitutionPDF';
import { uneAssociation } from '../constructeurs/constructeurAssociation';

describe('Adaptateur de Restitution', () => {
  const adaptateurRestitution =
    new (class extends AdaptateurDeRestitution<Buffer> {
      protected genere(
        htmlRecommandations: Promise<ContenuHtml>[],
      ): Promise<Buffer> {
        return Promise.all(htmlRecommandations).then((htmls) => {
          const resultat: ContenuHtml[] = [];

          htmls.forEach((html) => {
            resultat.push(html);
          });

          return Buffer.from(JSON.stringify(resultat), 'utf-8');
        });
      }
      protected genereIndicateurs(
        indicateurs: Indicateurs | undefined,
      ): Promise<ContenuHtml> {
        const resultat: ContenuHtml = {
          corps: '',
          entete: 'entete indicateur',
          piedPage: 'piedPage indicateur',
        };

        Object.entries(indicateurs || {})?.forEach(
          ([thematique, indicateur]) => {
            resultat.corps += JSON.stringify({ thematique, indicateur });
          },
        );

        return Promise.resolve(resultat);
      }

      protected genereRecommandationsAnnexes(
        autresRecommandations: RecommandationPriorisee[] | undefined,
      ): Promise<ContenuHtml> {
        const resultat: ContenuHtml = {
          corps: '',
          entete: 'entete',
          piedPage: 'piedPage',
        };

        autresRecommandations?.forEach((reco) => {
          resultat.corps += JSON.stringify(reco);
        });

        return Promise.resolve(resultat);
      }

      protected genereRecommandationsPrioritaires(
        recommandationsPrioritaires: RecommandationPriorisee[] | undefined,
      ): Promise<ContenuHtml> {
        const resultat: ContenuHtml = {
          corps: '',
          entete: 'entete',
          piedPage: 'piedPage',
        };

        recommandationsPrioritaires?.forEach((reco) => {
          resultat.corps += JSON.stringify(reco);
        });

        return Promise.resolve(resultat);
      }
    })();

  const questions = uneListeDeQuestions()
    .dontLesLabelsSont(['q1', 'q2'])
    .avecLesReponsesPossiblesSuivantesAssociees([
      {
        libelle: 'reponse 11',
        association: uneAssociation()
          .avecIdentifiant('q1')
          .deNiveau1()
          .ayantPourValeurDIndice(0)
          .construis(),
      },
      {
        libelle: 'reponse 12',
        association: uneAssociation()
          .avecIdentifiant('q1')
          .deNiveau2()
          .ayantPourValeurDIndice(1)
          .construis(),
      },
      { libelle: 'reponse 13' },
      { libelle: 'reponse 14' },
      {
        libelle: 'reponse 21',
        association: uneAssociation()
          .avecIdentifiant('q2')
          .deNiveau1()
          .ayantPourValeurDIndice(0)
          .construis(),
      },
      {
        libelle: 'reponse 22',
        association: uneAssociation()
          .avecIdentifiant('q2')
          .deNiveau2()
          .ayantPourValeurDIndice(1)
          .construis(),
      },
      { libelle: 'reponse 23' },
      { libelle: 'reponse 24' },
    ])
    .construis();

  const tableauDeRecommandations = unTableauDeRecommandations()
    .avecLesRecommandations([
      { q1: { niveau1: 'reco 1', niveau2: 'reco 12', priorisation: 3 } },
      { q2: { niveau1: 'reco 2', niveau2: 'reco 22', priorisation: 2 } },
      { q3: { niveau1: 'reco 2', niveau2: 'reco 22', priorisation: 2 } },
      { q4: { niveau1: 'reco 2', niveau2: 'reco 22', priorisation: 2 } },
      { q5: { niveau1: 'reco 2', niveau2: 'reco 22', priorisation: 2 } },
      { q6: { niveau1: 'reco 2', niveau2: 'reco 22', priorisation: 2 } },
      { q7: { niveau1: 'reco 2', niveau2: 'reco 22', priorisation: 2 } },
    ])
    .construis();

  it('génère la restitution sans annexe', async () => {
    const diagnostic = unDiagnostic()
      .avecUnReferentiel(
        unReferentiel()
          .ajouteUneThematique('thematique', questions)
          .construis(),
      )
      .avecLesReponsesDonnees('contexte', [{ qc: 'rqc' }])
      .avecLesReponsesDonnees('thematique', [
        { q2: 'reponse-23' },
        { q1: 'reponse-11' },
      ])
      .avecUnTableauDeRecommandations(tableauDeRecommandations)
      .construis();

    genereLaRestitution(diagnostic);

    expect(
      JSON.parse(
        (await adaptateurRestitution.genereRestitution(diagnostic)).toString(),
      ),
    ).toMatchSnapshot();
  });

  it('génère la restitution avec annexe', async () => {
    const questionsSupplementaires = uneListeDeQuestions()
      .dontLesLabelsSont(['q3', 'q4', 'q5', 'q6', 'q7'])
      .avecLesReponsesPossiblesSuivantesAssociees([
        {
          libelle: 'reponse 31',
          association: uneAssociation()
            .avecIdentifiant('q3')
            .deNiveau1()
            .ayantPourValeurDIndice(0)
            .construis(),
        },
        {
          libelle: 'reponse 41',
          association: uneAssociation()
            .avecIdentifiant('q4')
            .deNiveau1()
            .ayantPourValeurDIndice(0)
            .construis(),
        },
        {
          libelle: 'reponse 51',
          association: uneAssociation()
            .avecIdentifiant('q5')
            .deNiveau1()
            .ayantPourValeurDIndice(0)
            .construis(),
        },
        {
          libelle: 'reponse 61',
          association: uneAssociation()
            .avecIdentifiant('q6')
            .deNiveau1()
            .ayantPourValeurDIndice(0)
            .construis(),
        },
        {
          libelle: 'reponse 71',
          association: uneAssociation()
            .avecIdentifiant('q7')
            .deNiveau1()
            .ayantPourValeurDIndice(0)
            .construis(),
        },
      ])
      .construis();
    const diagnostic = unDiagnostic()
      .avecUnReferentiel(
        unReferentiel()
          .ajouteUneThematique('thematique', [
            ...questions,
            ...questionsSupplementaires,
          ])
          .construis(),
      )
      .avecLesReponsesDonnees('contexte', [{ qc: 'rqc' }])
      .avecLesReponsesDonnees('thematique', [
        { q1: 'reponse-11' },
        { q2: 'reponse-21' },
        { q3: 'reponse-31' },
        { q4: 'reponse-41' },
        { q5: 'reponse-51' },
        { q6: 'reponse-61' },
        { q7: 'reponse-71' },
      ])
      .avecUnTableauDeRecommandations(tableauDeRecommandations)
      .construis();

    genereLaRestitution(diagnostic);

    const buffer = await adaptateurRestitution.genereRestitution(diagnostic);

    buffer.toJSON();

    expect(
      JSON.parse(
        (await adaptateurRestitution.genereRestitution(diagnostic)).toString(),
      ),
    ).toMatchSnapshot();
  });
});
