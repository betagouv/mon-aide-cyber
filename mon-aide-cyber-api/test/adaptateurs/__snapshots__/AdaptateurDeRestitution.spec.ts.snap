// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Adaptateur de Restitution > génère la restitution avec annexe 1`] = `
[
  {
    "corps": "{\\"thematique\\":\\"thematique\\",\\"indicateur\\":{\\"moyennePonderee\\":0}}",
    "entete": "entete indicateur",
    "piedPage": "piedPage indicateur",
  },
  {
    "corps": "{\\"titre\\":\\"reco 2\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":2,\\"valeurObtenue\\":0}{\\"titre\\":\\"reco 2\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":2,\\"valeurObtenue\\":0}{\\"titre\\":\\"reco 2\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":2,\\"valeurObtenue\\":0}{\\"titre\\":\\"reco 2\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":2,\\"valeurObtenue\\":0}{\\"titre\\":\\"reco 2\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":2,\\"valeurObtenue\\":0}{\\"titre\\":\\"reco 2\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":2,\\"valeurObtenue\\":0}",
    "entete": "entete",
    "piedPage": "piedPage",
  },
  {
    "corps": "{\\"titre\\":\\"reco 1\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":3,\\"valeurObtenue\\":0}",
    "entete": "entete",
    "piedPage": "piedPage",
  },
]
`;

exports[`Adaptateur de Restitution > génère la restitution sans annexe 1`] = `
[
  {
    "corps": "{\\"thematique\\":\\"thematique\\",\\"indicateur\\":{\\"moyennePonderee\\":0}}",
    "entete": "entete indicateur",
    "piedPage": "piedPage indicateur",
  },
  {
    "corps": "{\\"titre\\":\\"reco 1\\",\\"pourquoi\\":\\"parce-que\\",\\"comment\\":\\"comme ça\\",\\"priorisation\\":3,\\"valeurObtenue\\":0}",
    "entete": "entete",
    "piedPage": "piedPage",
  },
]
`;
